### PR TITLE
Initial version of contributing guide

### PR DESCRIPTION
**Pull request by jdye64**

### CODE REVIEW

**Comment by jdye64** [comment]
Wanted to start a document to assist with Rust development. Decided it best that since we don't have a `CONTRIBUTING.md` file this was a good time to add one. This is a living document and would love others to help fill it up.

My Miro subscription ended as I was making the sequence diagram so that is the reason for the awkward and unfinished diagram image. Apologies for that.

**Review by charlesbluca — COMMENTED** [review]
Thanks for getting this started @jdye64 😄 a couple early comments around what you have so far, happy to contribute and make my own modifications here as well (if you're okay with that) since this is a pretty large undertaking.

Also I notice that you've made a new folder `assets` for the image used here - generally our GitHub page assets are stored in the `.github` folder, so that should probably be moved there (though I wouldn't be opposed to moving all of our assets to a dedicated folder later on)

**Comment by jdye64** [comment]
@charlesbluca yes please! Make all the changes you want! I just wanted to get something started. I'm on vacation for awhile so do with this as you please

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+### Building
+Building the Dask-SQL Rust codebase is a straightforward process. If you create and activate the Dask-SQL Conda environment the Rust compiler and all necessary components will be installed for you during that process and therefore requires no further manual setup.
+
+`setuptools-rust` is used by Dask-SQL for building and bundling the resulting Rust binaries. This helps make building and installing the Rust binaries feel much more like a native Python workflow.
+
+More details about the building setup can be found at [setup.py](setup.py) and searching for `rust_extensions` which is the hook for the Rust code build and inclusion.
+
+Note that while `setuptools-rust` is used by CI and should be used during your development cycle, if the need arises to do something more specific that is not yet supported by `setuptools-rust` you can opt to use `cargo` directly from the command line.
```
[the remark this answers is not in the document]
Comment (jdye64) [reply CONTRIBUTING.md:32]: It would be. Would you mind if we did that in a follow up PR however?

**Pull request merged**